From 1d2ceb430a38e68e05d15006a4b3121861ae82e9 Mon Sep 17 00:00:00 2001
From: Ricardo Salveti <ricardo@foundries.io>
Date: Tue, 30 Aug 2022 13:26:44 -0300
Subject: [PATCH] versal: add option to disable AES-GCM unaligned tests

Add CFG_CRYPTO_VERSAL build option to allow disabling AES-GCM unaligned
tests as they are known to be broken at this stage.

Upstream-Status: Inappropriate [lmp specific]

Signed-off-by: Ricardo Salveti <ricardo@foundries.io>
---
 host/xtest/CMakeLists.txt    | 4 ++++
 host/xtest/Makefile          | 4 ++++
 host/xtest/regression_4000.c | 2 ++
 3 files changed, 10 insertions(+)

diff --git a/host/xtest/CMakeLists.txt b/host/xtest/CMakeLists.txt
index 040043a..bfb8250 100644
--- a/host/xtest/CMakeLists.txt
+++ b/host/xtest/CMakeLists.txt
@@ -113,6 +113,10 @@ if (CFG_CRYPTO_SE05X)
 	add_compile_options(-DCFG_CRYPTO_SE05X)
 endif()
 
+if (CFG_CRYPTO_VERSAL)
+	add_compile_options(-DCFG_CRYPTO_VERSAL)
+endif()
+
 ################################################################################
 # Built binary
 ################################################################################
diff --git a/host/xtest/Makefile b/host/xtest/Makefile
index a898c57..a3ffd17 100644
--- a/host/xtest/Makefile
+++ b/host/xtest/Makefile
@@ -105,6 +105,10 @@ ifeq ($(CFG_CRYPTO_SE05X),y)
 CFLAGS += -DCFG_CRYPTO_SE05X
 endif
 
+ifeq ($(CFG_CRYPTO_VERSAL),y)
+CFLAGS += -DCFG_CRYPTO_VERSAL
+endif
+
 CFLAGS += -I./
 CFLAGS += -I./adbg/include
 CFLAGS += -I../supp_plugin/include
diff --git a/host/xtest/regression_4000.c b/host/xtest/regression_4000.c
index b03c6ab..fd75049 100644
--- a/host/xtest/regression_4000.c
+++ b/host/xtest/regression_4000.c
@@ -2427,6 +2427,7 @@ static const struct xtest_ae_case ae_cases[] = {
 	XTEST_AE_CASE_AES_CCM(vect2, 7, 13),
 	XTEST_AE_CASE_AES_CCM(vect3, 5, 21),
 
+#ifndef CFG_CRYPTO_VERSAL
 	XTEST_AE_CASE_AES_GCM(vect1, 0, 0, NULL_ARRAY, NULL_ARRAY, NULL_ARRAY),
 	XTEST_AE_CASE_AES_GCM(vect2, 0, 9, NULL_ARRAY, ARRAY, ARRAY),
 	XTEST_AE_CASE_AES_GCM(vect3, 0, 9, NULL_ARRAY, ARRAY, ARRAY),
@@ -2446,6 +2447,7 @@ static const struct xtest_ae_case ae_cases[] = {
 	XTEST_AE_CASE_AES_GCM(vect16, 5, 9, ARRAY, ARRAY, ARRAY),
 	XTEST_AE_CASE_AES_GCM(vect17, 5, 9, ARRAY, ARRAY, ARRAY),
 	XTEST_AE_CASE_AES_GCM(vect18, 5, 9, ARRAY, ARRAY, ARRAY),
+#endif
 #ifdef CFG_GCM_NIST_VECTORS
 #include "gcmDecrypt128.h"
 #include "gcmDecrypt192.h"
-- 
2.37.1

